lib/utils/common-flags.ts

```diff
@@ -87,3 +90,42 @@ export const json: IBooleanFlag<boolean> = flags.boolean({
 	description: 'produce JSON output instead of tabular output',
 	default: false,
 });
+
+export const dataOutputFlags: flags.Input<DataOutputOptions> = {
+	fields: flags.string({
+		exclusive: ['extended'],
+		description: 'only show provided fields (comma-separated)',
+	}),
+	json: flags.boolean({
+		exclusive: ['no-truncate'],
+		description: 'output in json format',
+		default: false,
+	}),
+	// Not yet supported.
+	// verbose: flags.boolean({
+	// 	exclusive: ['fields'],
+	// 	description: 'show extra fields',
+	// 	default: false,
+	// }),
+};
+
+export const dataSetOutputFlags: flags.Input<DataOutputOptions> &
+	flags.Input<DataSetOutputOptions> = {
+	...dataOutputFlags,
+	filter: flags.string({
+		description: 'filter results by partial string match, eg: name=foo',
```

Comment:
Would it be clearer to say: "filter results that match the given field=value pair (substring matching)" (maybe)

And should this flag be `multiple: true`?... (In case cli-ux allows an array of filters.)

Reply:
> And should this flag be multiple: true?... (In case cli-ux allows an array of filters.)

Did some tests, and it seems it only supports a single filter

Reply:
> Would it be clearer to say: "filter results that match the given field=value pair (substring matching)" (maybe)

Hmm, not sure.  How about?:

"filter results by field value (partial string match)"

Reply:
I think we need to add sample usage, either like you had done: `eg: name=foo'`, or in the Examples section of every CLI command that accepts `--filter`. Without examples, the usage of `--filter` is not obvious. "Partial string match" sounds ambiguous to me -- it could mean substring matching, or it could mean that one word out of two words matched, like a web search (e.g. `--filter "name=raspberry jetson"` might match fields that include either "raspberry" or "jetson"). I know that it means "substring matching" because I saw the examples in the PR description (main GitHut PR page, Conversation tab). So I think that a clear description would mention `name=value` or `field=value`, and mention "substring", ideally all in a single line... :-)
Perhaps a smaller variation of the original version:
```
'filter results by substring matching of a given field, eg: --filter field=foo'
```